src/Catalog.API/Apis/CatalogApi.cs

```diff
@@ -10,30 +13,85 @@ public static IEndpointRouteBuilder MapCatalogApiV1(this IEndpointRouteBuilder a
+            .WithName("ListItems")
+            .WithSummary("List catalog items")
+            .WithDescription("Get a paginated list of items in the catalog.")
+            .WithTags("Items");
+        api.MapGet("/items/by", GetItemsByIds)
+            .WithName("BatchGetItems")
+            .WithSummary("Batch get catalog items")
+            .WithDescription("Get multiple items from the catalog")
+            .WithTags("Items");
+        api.MapGet("/items/{id:int}", GetItemById)
+            .WithName("GetItem")
+            .WithSummary("Get catalog item")
+            .WithDescription("Get an item from the catalog")
+            .WithTags("Items");
+        api.MapGet("/items/by/{name:minlength(1)}", GetItemsByName)
+            .WithName("GetItemsByName")
+            .WithSummary("Get catalog items by name")
+            .WithDescription("Get a paginated list of catalog items with the specified name.")
+            .WithTags("Items");
+        api.MapGet("/items/{id:int}/pic", GetItemPictureById)
+            .WithName("GetItemPicture")
+            .WithSummary("Get catalog item picture")
+            .WithDescription("Get the picture for a catalog item")
+            .WithTags("Items");
 
         // Routes for resolving catalog items using AI.
-        api.MapGet("/items/withsemanticrelevance/{text:minlength(1)}", GetItemsBySemanticRelevance);
+        api.MapGet("/items/withsemanticrelevance/{text:minlength(1)}", GetItemsBySemanticRelevance)
+            .WithName("GetRelevantItems")
+            .WithSummary("Search catalog for relevant items")
+            .WithDescription("Search the catalog for items related to the specified text")
+            .WithTags("Search");
 
         // Routes for resolving catalog items by type and brand.
-        api.MapGet("/items/type/{typeId}/brand/{brandId?}", GetItemsByBrandAndTypeId);
-        api.MapGet("/items/type/all/brand/{brandId:int?}", GetItemsByBrandId);
-        api.MapGet("/catalogtypes", async (CatalogContext context) => await context.CatalogTypes.OrderBy(x => x.Type).ToListAsync());
-        api.MapGet("/catalogbrands", async (CatalogContext context) => await context.CatalogBrands.OrderBy(x => x.Brand).ToListAsync());
+        api.MapGet("/items/type/{typeId}/brand/{brandId?}", GetItemsByBrandAndTypeId)
+            .WithName("GetItemsByTypeAndBrand")
+            .WithSummary("Get catalog items by type and brand")
+            .WithDescription("Get catalog items of the specified type and brand")
+            .WithTags("Types");
+        api.MapGet("/items/type/all/brand/{brandId:int?}", GetItemsByBrandId)
+            .WithName("GetItemsByBrand")
+            .WithSummary("List catalog items by brand")
+            .WithDescription("Get a list of catalog items for the specified brand")
+            .WithTags("Brands");
+        api.MapGet("/catalogtypes",
+            [ProducesResponseType<ProblemDetails>(StatusCodes.Status400BadRequest, "application/problem+json")]
```

Comment:
Remind me again why we're not using `.ProducesProblem(StatusCodes.Status400BadRequest, "application/problem+json")`?